recipes/crashpad/all/test_package/conanfile.py

```diff
@@ -1,25 +1,39 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
+from conan.tools.files import mkdir, save, load
 import os
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        save(self, os.path.join(self.generators_folder, "bindir"),
+             os.path.join(self.dependencies["crashpad"].package_folder, "bin"))
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
+        if can_run(self):
             test_env_dir = "test_env"
-            tools.mkdir(test_env_dir)
-            bin_path = os.path.join("bin", "test_package")
-            handler_exe = "crashpad_handler.exe" if self.settings.os == "Windows" else "crashpad_handler"
-            handler_bin_path = os.path.join(self.deps_cpp_info["crashpad"].rootpath, "bin", handler_exe)
-            self.run("%s %s/db %s" % (bin_path, test_env_dir, handler_bin_path), run_environment=True)
+            mkdir(self, test_env_dir)
+            bin_path = os.path.join(self.cpp.build.bindir, "test_package")
+            bindir = load(self, os.path.join(self.generators_folder, "bindir"))
+            handler_exe = "crashpad_handler" + (".exe" if self.settings.os == "Windows" else "")
+            handler_bin_path = os.path.join(bindir, handler_exe)
+            self.run(f"{bin_path} {test_env_dir}/db {handler_bin_path}", env="conanrun")
```

Comment:
The code that this is building is way too much for my linking, it even creates a localhost listener. I want to gets this PR merged soon, so won't block on it, but it would be nice to come back to the recipe to clean the test package up :)